onyo/lib/tests/test_git.py

```diff
@@ -0,0 +1,264 @@
+from pathlib import Path
+
+import pytest
+from onyo import OnyoRepo, OnyoInvalidRepoError
+from onyo.lib.git import GitRepo
+
+
+def test_GitRepo_instantiation_existing(repo: OnyoRepo) -> None:
+    """
+    The GitRepo class must instantiate correctly for paths to existing
+    repositories.
+    """
+    new_repo = GitRepo(repo.git.root)
+    assert new_repo.root.samefile(repo.git.root)
+
+
+@pytest.mark.repo_dirs('existing/directory')
+def test_GitRepo_instantiation_find_root(repo: OnyoRepo) -> None:
+    """
+    The GitRepo class must instantiate correctly with paths to sub-directories
+    in a repository when `find_root=True`.
+    """
+    new_repo = GitRepo(repo.git.root / 'existing/directory', find_root=True)
+    assert new_repo.root.samefile(repo.git.root)
+    assert (new_repo.root / '.git').exists()
+
+
+def test_GitRepo_find_root(tmp_path: Path) -> None:
```

Comment:
So, generally we are aiming at unit tests here. Meaning that any method (`GitRepo.find_root` in this case) should come up with a test, that defines its behavior. (On that note: I'd also fuse the instantiation tests into one. One test that tells us, whether `GitRepo`'s instantiation is broken or not).

Here you only have half of it. Two cases of failure. No case of success, though.